calendar/modules/listener/http_service.bal

```diff
@@ -59,51 +58,69 @@ service class HttpService {
-                if (created is string && updated is string && 'start is calendar:Time && end is calendar:Time) {
-                    if (created.substring(0, 19) == updated.substring(0, 19)) {
-                       if (self.isOnNewEventAvailable) {
-                           check callOnNewEventMethod(self.httpService, event);
-                       }
-                    } else  {
-                        if (self.isOnEventUpdateAvailable) {
-                            check callOnEventUpdateMethod(self.httpService, event);
-                        }
-                    }
-                } else {
-                    if (self.isOnEventDeleteAvailable) {
-                        check callOnEventDeleteMethod(self.httpService, event);
-                    }
-                }
             }
+        } else {
+            return error(INVALID_ID_ERROR);
         }
-        return error (INVALID_ID_ERROR);
     }
 
-    isolated function getNextPageToken(calendar:Client httpClient, string calendarId, string? pageToken = ()) returns
-        @tainted string?|error {
+    isolated function validateRequest(http:Request request) returns boolean|error {
+        return ((check request.getHeader(GOOGLE_CHANNEL_ID)) == self.channelId && (check request.getHeader(
+        GOOGLE_RESOURCE_ID)) == self.resourceId);
+    }
+
+    isolated function getInitialSyncToken(calendar:Client httpClient, string calendarId, string? pageToken = ()) 
+    returns @tainted string?|error {
         calendar:EventResponse resp = check httpClient->getEventsResponse(calendarId, pageToken = pageToken);
         string? nextPageToken = resp?.nextPageToken;
+        string? syncToken = ();
         if (nextPageToken is string) {
-            var token = check self.getNextPageToken(httpClient, calendarId, nextPageToken);
+            syncToken = check self.getInitialSyncToken(httpClient, calendarId, nextPageToken);
+        }
+        if (syncToken is string) {
+            return syncToken;
         }
         return resp?.nextSyncToken;
     }
+
+    isolated function validateSyncRequest(http:Request request) returns boolean|error {
+        return ((check request.getHeader(GOOGLE_RESOURCE_STATE)) == SYNC);
+    }
+
+    isolated function processEvent() returns [string, calendar:Event]|error {
+        calendar:EventResponse resp = check self.calendarClient->getEventsResponse(self.calendarId, syncToken = self.
+        currentSyncToken);
+        string syncToken = resp?.nextSyncToken ?: "";
+        calendar:Event event = resp?.items[0];
+        return [syncToken, event];
+    }
+
+    isolated function dispatchEvent(calendar:Event event) returns error? {
+        string? created = event?.created;
+        string? updated = event?.updated;
+        calendar:Time? 'start = event?.'start;
+        calendar:Time? end = event?.end;
+        if (created is string && updated is string && 'start is calendar:Time && end is calendar:Time) {
```

Comment:
Shall we add another method called isCreateOrUpdateEvent() and evaluate this logic there?

Reply:
yes. Fixed in https://github.com/ballerina-platform/module-ballerinax-googleapis.calendar/commit/434fe847800ecdb8c18b17829934a78d7232af61